dwave/cloud/__init__.py

```diff
@@ -35,11 +34,36 @@
 configure_logging_from_env(logger)
 
 
-# alias dwave.cloud.client.{qpu,sw,hybrid} as dwave.cloud.*
+# lazily import Client/Solver/Future -- only when actually asked for
+def __getattr__(cls):
```

Comment:
Should we implement [`__dir__()`](https://docs.python.org/3/reference/datamodel.html#object.__dir__) as well? Might be anecdotal, but I use `dir(module)` fairly often when using a new library.

Reply:
I can add `__dir__()` but I think I would like to deprecate `from dwave.cloud import Client` form. This lazy getattr import is here to support the existing code, but I'm no sure I want to promote it further. :shrug:

Reply:
Fair enough. I do think at least `__dir__()` and `__all__` should be in sync, but flexible :shrug:

Reply:
Fair point, I'll add it.

Reply:
Added `__dir_()`. But this note in [docs](https://docs.python.org/3/library/functions.html#dir) is interesting:
> Note: Because [dir()](https://docs.python.org/3/library/functions.html#dir) is supplied primarily as a convenience for use > at an interactive prompt, it tries to supply an interesting set of names more than it tries to supply a rigorously or
> consistently defined set of names, and its detailed behavior may change across releases. For example, metaclass
> attributes are not in the result list when the argument is a class.

Reply:
For sure. I would never treat it as authoritative. I do think it's a useful convenience feature, in a prompt for instance, so nice to have.